tasks/linux.yml

```diff
@@ -0,0 +1,11 @@
+- name: Enable WireGuard kernel module
```

Comment:
You don't need an extra file. Leave it were it's currently and put it in a `block:` with a condition `when: ansible_os_is_linux`. Here is an example how this looks like: https://github.com/githubixx/ansible-role-wireguard/blob/master/tasks/main.yml#L55-L68